### ISSUE TITLE
[Announcement] 4.0.0-preview1 is now available

### ISSUE THREAD

**Issue by Tratcher** [aspnet/AspNetKatana#140]
4.0.0-preview1 is now available. We anticipate this will be the last prerelease. If you like it we'll ship the final version in a few weeks.
https://www.nuget.org/packages/Microsoft.Owin/4.0.0-preview1

This includes updated IdentityModel preview2 dependencies with WsFederation bug fixes. It also includes a few local bug fixes: 
https://github.com/aspnet/aspnetkatana/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20milestone%3A4.0.0-preview1%20

**Comment by Tratcher** [comment]
https://www.nuget.org/packages/Microsoft.Owin/4.0.0 is now available. Enjoy 😀

**Comment by snekbaev** [comment]
Finally! :) Will upgrade during the daytime :)

**Comment by rickbutton** [comment]
Is there any documentation on compatibility? I can't seem to find any docs for 4.0.0, or anything for moving from 3.0.0 -> 4.0.0.

**Comment by Tratcher** [comment]
@rickbutton See https://github.com/aspnet/AspNetKatana/wiki/Roadmap#400-release-february-2018

The main compat issue is if you were using any of the IdentityModel types directly, like SecurityTokenValidator.

**Comment by rabrahamOpti** [comment]
@Tratcher Can you point to documentation that lists all the breaking changes, and the workarounds for those changes?

**Comment by Tratcher** [comment]
@rabrahamOpti See https://github.com/aspnet/AspNetKatana/wiki/Roadmap#400-release-february-2018

Specific guidance should not be needed for most users, only users that made extensive use the the IdentityModel types like TokenValidationParameters will need to change code.

**Comment by rabrahamOpti** [comment]
@Tratcher I'm updating 3.0.0 -> 4.0.0 in a code base I'm not familiar with. It looks like they're creating a  `Microsoft.Owin.Security.Jwt.JwtFormat`, passing in a `TokenValidationParameters` object, and a `IIssuerSecurityTokenProvider` object. The constructor for `JwtFormat` seems to have changed to accept a `IIssuerSecurityKeyProvider` instead. I'm looking for documentation on the relationship between these two interfaces. Has `IIssuerSecurityTokenProvider` been deprecated?

**Comment by Tratcher** [comment]
Yes, everything that was using SecurityTokens is now using SecurityKeys. What kind of provider were they using? There should be a one to one replacement.
https://github.com/aspnet/AspNetKatana/commit/5d58dab237663b694515a3adbb36dd246a7d4ca2#diff-02dcc6bdd8fadc8c97e5234d641548fcR13
